### ISSUE TITLE
(feature request) Reexporting default exports as ${fileName}

### ISSUE THREAD

**Issue by hasparus** [testpossessed/auto-barrel#1]
Hi! First things first -- I wanted to say I really like the extension. 

I think it would be really useful in projects that use `default` export, because writing
```js
export { default as foo } from './foo';
```
gets super tedious.

We'd probably need sth like
```ts
if (reexportDefaultAsFileName) {
  workspaceEdit.insert(fileUri, position, `export { default as ${fileName} } from './${fileName}';\n`);
}
```
after line 45.

I could create a PR if you like the idea 😉

**Comment by testpossessed** [comment]
Hi Piotr,

Glad you like the extension.  It certainly helps me be more productive.

Please do create a PR, it will help clear up the idea for me.  I have never used that form of export so not familiar with the problem it solves, but sounds like something that we should have.

Mike

**Comment by hasparus** [comment]
Lots of people like to do `export default X`.
And as I'm aware `export foo from './foo'` doesn't work and to have these default exports reexported in a barrel we have to do `export { default as foo } from './foo'`. 

The problem is there's no autocomplete on the first `foo`, because that's a new identifier (that's why avoiding default exports is better IMO) so doing this manually is not only tedious but error prone for long names.

I'm gonna create a PR this week (I'd like it to be tomorrow but probably can't promise that :D)

**Comment by testpossessed** [comment]
Understood, I can have a go at it today.

I guess the flag comes from scanning the file we are pulling into the barrel for 'export default..' and if that is the case use the appropriate export statement?

**Comment by testpossessed** [comment]
So I have just pushed and published a change to include this feature.

Works fine in Create Barrel command on folder that contains a file containing "export default..."

I added support into the start command handling too, but it only really applies when an existing file is moved into the barrel folder.

The situation we don't handle is a new file being added then the developer adds a default export.

I am not sure auto-barrel should go as far as watching files for this sort of change and updating the barrel.
